plugin/test-keeper/plugin/test_matcher.go

```diff
@@ -0,0 +1,56 @@
+package plugin
+
+import (
+	"strings"
+	"fmt"
+	"regexp"
+	"gopkg.in/yaml.v2"
+)
+
+type TestMatcher struct {
+	TestRegex string `yaml:"tests_pattern"`
+}
+
+var defaultMatcher = TestMatcher{
+	TestRegex: `.*[tT]est.*`,
+}
+
+var javaMatcher = TestMatcher{
+	TestRegex: `.+((IT|Test|TestCase)\.java)$`,
+}
+
+var goMatcher = TestMatcher{
+	TestRegex: `.*(_test.go)$`,
+}
+
+var langMatchers = map[string]TestMatcher{
+	"java": javaMatcher,
+	"go":   goMatcher,
+}
+
+func (matcher *TestMatcher) isTest(file string) bool {
+	return regexp.MustCompile(matcher.TestRegex).MatchString(file)
+}
+
+func loadMatcherFromConfig(content []byte) TestMatcher {
+	var matcher TestMatcher
+	yaml.Unmarshal(content, &matcher)
```

Comment:
Are we fine with ignoring errors?

Reply:
I wouldn't fail, but you are right that totally ignoring them is not the best - I'll log the possible errors